src/status_im/subs.cljs

```diff
@@ -969,21 +993,39 @@
  (fn [[_ chat-id] _]
    [(re-frame/subscribe [:chats/message-list chat-id])
     (re-frame/subscribe [:chats/chat-messages chat-id])
+    (re-frame/subscribe [:chats/pinned chat-id])
     (re-frame/subscribe [:chats/loading-messages? chat-id])
     (re-frame/subscribe [:chats/synced-from chat-id])
     (re-frame/subscribe [:chats/chat-type chat-id])
     (re-frame/subscribe [:chats/joined chat-id])])
- (fn [[message-list messages loading-messages? synced-from chat-type joined] [_ chat-id]]
+ (fn [[message-list messages pin-messages loading-messages? synced-from chat-type joined] [_ chat-id]]
    ;;TODO (perf)
    (let [message-list-seq (models.message-list/->seq message-list)]
      ; Don't show gaps if that's the case as we are still loading messages
      (if (and (empty? message-list-seq) loading-messages?)
        []
        (-> message-list-seq
            (chat.db/add-datemarks)
-           (hydrate-messages messages)
+           (hydrate-messages messages pin-messages)
            (chat.db/collapse-gaps chat-id synced-from (datetime/timestamp) chat-type joined loading-messages?))))))
 
+(re-frame/reg-sub
+ :chats/raw-chat-pin-messages-stream
+ (fn [[_ chat-id] _]
+   [(re-frame/subscribe [:chats/pin-message-list chat-id])
```

Comment:
It is a completely valid question, right now it is possible to see pinned messages without having messages from chat loaded (i.e accessing from user profile to pinned messages), so it can be possible to not have the message loaded and get the pinned messages from json-rpc API (which also provides some additional properties such as `pinned-by`, `pinned-at` and `pinned`. Also there are some diffs on params names which can conflict (message_id is one example, and on receiving signals they come with some differences too)